### PR TITLE
Feature/bhr 110 edit crisis templates

### PR DESCRIPTION
**Pull request by solomontang**
Add edit functionality to the crisis template.

### CODE REVIEW

File: src/components/crisis/SubmitSuccess.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+`;
+
+type Props = {
+
+};
+
+class SubmitSuccess extends Component<Props> {
+
+  clearAndNavigate = (path :string) => () => {
+
```
Comment (katherinebernstein) [inline src/components/crisis/SubmitSuccess.js]: is this supposed to do something?

File: src/components/modals/DeleteModal.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import Modal from '@atlaskit/modal-dialog';
+
+type Props = {
+  onClose :() => void;
+  actions :Object[];
+};
+
+const DeleteModal = ({ actions, onClose } :Props) => {
+
+  const message = 'Clicking on "Delete Temmplate" will delete the current Crisis Template and return to the home screen.';
```
Comment (katherinebernstein) [inline src/components/modals/DeleteModal.js]: Typo on "Template"

File: src/components/modals/DeleteModal.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <Modal
+        actions={actions}
+        heading="Delete Temmplate?"
```
Comment (katherinebernstein) [inline src/components/modals/DeleteModal.js]: same typo

File: src/containers/pages/subjectinformation/SubjectQuickSearch.js [hunk 4]
```diff
@@ -115,7 +116,7 @@ class SubjectQuickSearch extends Component<Props, State> {
       [SUBJECT_INFORMATION.RACE]: person.getIn([PERSON_RACE_FQN, 0], ''),
       [SUBJECT_INFORMATION.GENDER]: person.getIn([PERSON_SEX_FQN, 0], ''),
       [SUBJECT_INFORMATION.AGE]: moment().diff(moment(person.getIn([PERSON_DOB_FQN, 0], '')), 'years'),
-      [SUBJECT_INFORMATION.SSN_LAST_4]: 'XXXX',
+      [SUBJECT_INFORMATION.SSN_LAST_4]: person.getIn([PERSON_SSN_LAST_4_FQN, 0], ''),
```
Comment (katherinebernstein) [inline src/containers/pages/subjectinformation/SubjectQuickSearch.js]: If we're obfuscating the actual last 4 of the SSN when loading existing people, it seems sufficient to just make that box disabled, so we don't actually need this

File: src/containers/reports/CrisisReportView.js [hunk 5]
```diff
@@ -204,28 +237,129 @@ class CrisisTemplateContainer extends React.Component<Props> {
...
+  }
+
+  handleDiscard = () => {
+    const { actions, history } = this.props;
+    actions.clearCrisisTemplate();
+    history.push(HOME_PATH);
+  }
+
+  handleDelete = () => {
+    console.log('Doing delete things');
```
Comment (katherinebernstein) [inline src/containers/reports/CrisisReportView.js]: is this doing delete things?

File: src/containers/reports/ReportsReducer.js [hunk 6]
```diff
@@ -36,6 +32,10 @@ export default function reportReducer(state :Map<*, *> = INITIAL_STATE, action :
 
   switch (action.type) {
 
+    case CLEAR_CRISIS_TEMPLATE: {
+      return state;
```
Comment (katherinebernstein) [inline src/containers/reports/ReportsReducer.js]: is this supposed to return INITIAL_STATE ?

File: src/containers/reports/ReportsSagas.js [hunk 7]
```diff
@@ -340,26 +290,43 @@ function* updateReportWorker(action :SequenceAction) :Generator<*, *, *> {
...
   try {
     yield put(updateReport.request(action.id, value));
 
     const edm :Map<*, *> = yield select(state => state.get('edm'));
-    const entityData :Map<string, List<any>> = Map().withMutations((map :Map<string, List<any>>) => {
-      reportEdits.forEach((data :List<any>, fqn :FullyQualifiedName) => {
-        const propertyTypeId :string = edm.getIn(['fqnToIdMap', fqn]);
-        map.setIn([entityKeyId, propertyTypeId], data);
-      });
+    const app = yield select(state => state.getIn(['app'], Map()));
```
Comment (katherinebernstein) [inline src/containers/reports/ReportsSagas.js]: unimportant but you don't need to use getIn here -- seems a teensy bit cleaner to use .get and avoid the extra brackets

File: src/containers/reports/CrisisReportView.js (hunk 5 above)
Comment (solomontang) [reply src/containers/reports/CrisisReportView.js]: Delete is gonna come in another PR. I can remove the console.log for now tho

File: src/containers/pages/subjectinformation/SubjectQuickSearch.js (hunk 4 above)
Comment (solomontang) [reply src/containers/pages/subjectinformation/SubjectQuickSearch.js]: I've added type="password" to this field so the digits will be replaced by the browser's native password symbol.

File: src/containers/reports/ReportsReducer.js (hunk 6 above)
Comment (solomontang) [reply src/containers/reports/ReportsReducer.js]: Indeed. Thanks for catching that.

File: src/containers/reports/ReportsSagas.js (hunk 7 above)
Comment (solomontang) [reply src/containers/reports/ReportsSagas.js]: tyty

File: src/components/crisis/SubmitSuccess.js (hunk 1 above)
Comment (solomontang) [reply src/components/crisis/SubmitSuccess.js]: Fixed.

**Pull request merged**